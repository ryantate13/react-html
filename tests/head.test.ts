import {head} from '../src';

describe('head', () => {
    it('renders with children', () => {
        expect(head({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(head({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(head({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(head()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(head({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(head({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(head({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(head({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(head({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(head({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(head({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(head({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(head({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(head({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(head({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(head({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(head({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(head({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(head({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(head({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(head({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(head({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(head({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(head({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(head({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(head({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(head({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(head({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(head({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(head({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(head({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(head({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(head({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(head({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(head({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(head({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(head({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(head({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(head({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(head({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(head({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(head({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(head({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(head({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(head({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(head({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(head({translate: "no"})).toMatchSnapshot();
    });
});