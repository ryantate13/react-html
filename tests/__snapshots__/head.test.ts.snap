// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`head renders arrays of children 1`] = `
<head>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</head>
`;

exports[`head renders with accessKey="a,b,c,d" 1`] = `
<head
  accessKey="a,b,c,d"
/>
`;

exports[`head renders with autoCapitalize="characters" 1`] = `
<head
  autoCapitalize="characters"
/>
`;

exports[`head renders with autoCapitalize="none" 1`] = `
<head
  autoCapitalize="none"
/>
`;

exports[`head renders with autoCapitalize="off" 1`] = `
<head
  autoCapitalize="off"
/>
`;

exports[`head renders with autoCapitalize="on" 1`] = `
<head
  autoCapitalize="on"
/>
`;

exports[`head renders with autoCapitalize="sentences" 1`] = `
<head
  autoCapitalize="sentences"
/>
`;

exports[`head renders with autoCapitalize="words" 1`] = `
<head
  autoCapitalize="words"
/>
`;

exports[`head renders with children 1`] = `
<head>
  TEST
</head>
`;

exports[`head renders with className="foo bar baz" 1`] = `
<head
  className="foo bar baz"
/>
`;

exports[`head renders with contentEditable="false" 1`] = `
<head
  contentEditable="false"
/>
`;

exports[`head renders with contentEditable="true" 1`] = `
<head
  contentEditable="true"
/>
`;

exports[`head renders with dir="auto" 1`] = `
<head
  dir="auto"
/>
`;

exports[`head renders with dir="ltr" 1`] = `
<head
  dir="ltr"
/>
`;

exports[`head renders with dir="rtl" 1`] = `
<head
  dir="rtl"
/>
`;

exports[`head renders with draggable="false" 1`] = `
<head
  draggable="false"
/>
`;

exports[`head renders with draggable="true" 1`] = `
<head
  draggable="true"
/>
`;

exports[`head renders with empty attributes 1`] = `<head />`;

exports[`head renders with hidden=false 1`] = `
<head
  hidden={false}
/>
`;

exports[`head renders with hidden=true 1`] = `
<head
  hidden={true}
/>
`;

exports[`head renders with id="TEST" 1`] = `
<head
  id="TEST"
/>
`;

exports[`head renders with inputMode="decimal" 1`] = `
<head
  inputMode="decimal"
/>
`;

exports[`head renders with inputMode="email" 1`] = `
<head
  inputMode="email"
/>
`;

exports[`head renders with inputMode="none" 1`] = `
<head
  inputMode="none"
/>
`;

exports[`head renders with inputMode="numeric" 1`] = `
<head
  inputMode="numeric"
/>
`;

exports[`head renders with inputMode="search" 1`] = `
<head
  inputMode="search"
/>
`;

exports[`head renders with inputMode="tel" 1`] = `
<head
  inputMode="tel"
/>
`;

exports[`head renders with inputMode="text" 1`] = `
<head
  inputMode="text"
/>
`;

exports[`head renders with inputMode="url" 1`] = `
<head
  inputMode="url"
/>
`;

exports[`head renders with is="foo" 1`] = `
<head
  is="foo"
/>
`;

exports[`head renders with itemID="https://google.com/" 1`] = `
<head
  itemID="https://google.com/"
/>
`;

exports[`head renders with itemProp="https://google.com/" 1`] = `
<head
  itemProp="https://google.com/"
/>
`;

exports[`head renders with itemRef="foo bar" 1`] = `
<head
  itemRef="foo bar"
/>
`;

exports[`head renders with itemScope=false 1`] = `
<head
  itemScope={false}
/>
`;

exports[`head renders with itemScope=true 1`] = `
<head
  itemScope={true}
/>
`;

exports[`head renders with itemType="https://google.com/" 1`] = `
<head
  itemType="https://google.com/"
/>
`;

exports[`head renders with lang="" 1`] = `
<head
  lang=""
/>
`;

exports[`head renders with lang="es-419" 1`] = `
<head
  lang="es-419"
/>
`;

exports[`head renders with slot="TEST" 1`] = `
<head
  slot="TEST"
/>
`;

exports[`head renders with spellCheck="false" 1`] = `
<head
  spellCheck="false"
/>
`;

exports[`head renders with spellCheck="true" 1`] = `
<head
  spellCheck="true"
/>
`;

exports[`head renders with style={"color":"#000000"} 1`] = `
<head
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`head renders with tabIndex=1 1`] = `
<head
  tabIndex={1}
/>
`;

exports[`head renders with title="TEST" 1`] = `
<head
  title="TEST"
/>
`;

exports[`head renders with translate="no" 1`] = `
<head
  translate="no"
/>
`;

exports[`head renders with translate="yes" 1`] = `
<head
  translate="yes"
/>
`;

exports[`head renders without attributes 1`] = `<head />`;
